### PR TITLE
Adding isValidMotionProp

### PR DESCRIPTION
**Pull request by InventingWithMonster**

### CODE REVIEW

File: src/dom/valid-motion-props.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    "onHoverStart",
+])
+
+/**
+ * Check whether a prop name is a valid `MotionProp` key.
+ *
+ * @param key - Name of the property to check
+ * @returns `true` is key is a valid `MotionProp`.
+ *
+ * @public
```
Comment (nvh) [inline src/dom/valid-motion-props.ts]: Should this be part of the public api? I'm not sure. When would a consumer of the API use this?

File: src/motion/utils/valid-prop.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { MotionProps } from "../types"
```
Comment (nvh) [inline src/motion/utils/valid-prop.ts]: One of these files needs to be removed, I think

File: src/motion/utils/valid-props.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { MotionProps } from "../types"
```
Comment (nvh) [inline src/motion/utils/valid-props.ts]: One of these files needs to be removed, I think

File: src/dom/valid-motion-props.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import { MotionProps } from "../motion/types"
+
+/**
+ * A list of all valid MotionProps
+ *
+ * @internalremarks
+ * This doesn't throw if a `MotionProp` name is missing - it should.
+ */
+const validDomMotionProps = new Set<keyof MotionProps>([
```
Comment (nvh) [inline src/dom/valid-motion-props.ts]: What's the difference between this and validMotionProps?

File: src/dom/valid-motion-props.ts (hunk 1 above)
Comment (InventingWithMonster) [reply src/dom/valid-motion-props.ts]: Same thing we're doing, or `@emotion/is-prop-valid`

**Pull request merged**